inventory/agents_external_exporter_test.go

```diff
@@ -0,0 +1,293 @@
+			Body:    agents.GetAgentBody{AgentID: agentID},
+			Context: pmmapitests.Context,
+		})
+		require.NoError(t, err)
+		assert.Equal(t, &agents.GetAgentOKBody{
+			ExternalExporter: &agents.GetAgentOKBodyExternalExporter{
+				AgentID:      agentID,
+				ServiceID:    serviceID,
+				RunsOnNodeID: genericNodeID,
+				Username:     "username",
+				Scheme:       "https",
+				MetricPath:   "/metrics-hr",
+				ListenPort:   12345,
+				CustomLabels: map[string]string{
+					"custom_label_external_exporter": "external_exporter",
+				},
+			},
+		}, getAgentRes.Payload)
+
+		// Test change API.
+		changeExternalExporterOK, err := client.Default.Agents.ChangeExternalExporter(&agents.ChangeExternalExporterParams{
+			Body: agents.ChangeExternalExporterBody{
+				AgentID: agentID,
+				Common: &agents.ChangeExternalExporterParamsBodyCommon{
+					Disable:            true,
+					RemoveCustomLabels: true,
+				},
+			},
+			Context: pmmapitests.Context,
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, &agents.ChangeExternalExporterOKBody{
+			ExternalExporter: &agents.ChangeExternalExporterOKBodyExternalExporter{
+				AgentID:      agentID,
+				ServiceID:    serviceID,
+				RunsOnNodeID: genericNodeID,
+				Username:     "username",
+				Scheme:       "https",
+				MetricPath:   "/metrics-hr",
+				ListenPort:   12345,
+				Disabled:     true,
+			},
+		}, changeExternalExporterOK.Payload)
+
+		changeExternalExporterOK, err = client.Default.Agents.ChangeExternalExporter(&agents.ChangeExternalExporterParams{
+			Body: agents.ChangeExternalExporterBody{
+				AgentID: agentID,
+				Common: &agents.ChangeExternalExporterParamsBodyCommon{
+					Enable: true,
+					CustomLabels: map[string]string{
+						"new_label": "external_exporter",
+					},
+				},
+			},
+			Context: pmmapitests.Context,
+		})
+		assert.NoError(t, err)
+		assert.Equal(t, &agents.ChangeExternalExporterOKBody{
+			ExternalExporter: &agents.ChangeExternalExporterOKBodyExternalExporter{
+				AgentID:      agentID,
+				ServiceID:    serviceID,
+				RunsOnNodeID: genericNodeID,
+				Username:     "username",
+				Scheme:       "https",
+				MetricPath:   "/metrics-hr",
+				ListenPort:   12345,
+				Disabled:     false,
+				CustomLabels: map[string]string{
+					"new_label": "external_exporter",
+				},
+			},
+		}, changeExternalExporterOK.Payload)
+	})
+
+	t.Run("AddServiceIDEmpty", func(t *testing.T) {
+		t.Parallel()
+
+		genericNodeID := addGenericNode(t, pmmapitests.TestString(t, "")).NodeID
+		require.NotEmpty(t, genericNodeID)
+		defer pmmapitests.RemoveNodes(t, genericNodeID)
+
+		res, err := client.Default.Agents.AddExternalExporter(&agents.AddExternalExporterParams{
+			Body: agents.AddExternalExporterBody{
+				ServiceID:    "",
+				RunsOnNodeID: genericNodeID,
+				ListenPort:   12345,
+			},
+			Context: pmmapitests.Context,
+		})
+		pmmapitests.AssertAPIErrorf(t, err, 400, codes.InvalidArgument, "Empty Service ID.")
+		if !assert.Nil(t, res) {
```

Comment:
assert.NotNil

Reply:
No, `nil` it's what I expect to have here and if it's not nil I should remove service manually.